src/NUnitFramework/framework/Internal/Execution/TestWorker.cs

```diff
@@ -157,7 +178,21 @@ public void Start()
             _workerThread = new Thread(new ThreadStart(TestWorkerThreadProc));
             _workerThread.Name = Name;
 #if APARTMENT_STATE
-            _workerThread.SetApartmentState(WorkQueue.TargetApartment);
+#if NETSTANDARD2_0
+            if (!RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows))
+            {
+                if (WorkQueue.TargetApartment != ApartmentState.STA)
```

Comment:
Could you do this check the other way around (WorkQueue.TargetApartment property first, then `IsOSPlatform`)?